lib/spack/spack/directory_layout.py

```diff
@@ -190,9 +190,7 @@ def __init__(self, root, **kwargs):
         self.metadata_dir        = '.spack'
         self.spec_file_name      = 'spec.yaml'
         self.extension_file_name = 'extensions.yaml'
-        self.build_log_name      = 'build.txt'  # build log.
```

Comment:
As mentioned in the comment starting this block: older installations will store their build logs in this location, so if possible we would want a way to retrieve build log files by their old name (but still always generate them according to their new name).

Reply:
Should backward compatibility with `build.out` and `env.out` also be supported?

The current properties/methods neither check nor require that the log files actually exist so can be used in any build phase (i.e., whether the files exist or not).  So I presume the goal is to return the old name if the file exists; otherwise, use the new/current name.

Reply:
> I presume the goal is to return the old name if the file exists; otherwise, use the new/current name.

Yes

Reply:
@scheibelp Okay. but should I also be checking for `build.out` and `env.out` (for backward compatibility)?

Reply:
It wasn't that much extra work so I added checks for names before and after #11749 .